MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m

```diff
@@ -679,14 +679,19 @@ - (void)closeApp __attribute__((noreturn)) {
   exit(0);
 }
 
-- (void)openUrl:(NSURL *)url {
+- (BOOL)openUrl:(NSURL *)url {
+  
+  /*
+   * Ignore if app secret not set, can't test scheme.
+   * Also ignore if request is not for Mobile Center Distribute and this app.
+   */
+  if (!self.appSecret || ![[url scheme] isEqualToString:[NSString stringWithFormat:kMSDefaultCustomSchemeFormat, self.appSecret]]) {
```

Comment:
I agree that we need to check for more than the beginning as there could be more than 1 app on the device that has mobile center integrated.